engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationInputBuilder.kt

```diff
@@ -1,172 +1,159 @@
-import graphql.language.BooleanValue
-import graphql.language.FloatValue
-import graphql.language.IntValue
 import graphql.language.NullValue
-import graphql.language.ObjectField
-import graphql.language.ObjectValue
-import graphql.language.StringValue
-import graphql.language.Value
 import graphql.nadel.enginekt.blueprint.NadelHydrationFieldInstruction
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationActorInputDef
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationActorInputDef.ValueSource
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationStrategy
 import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
 import graphql.nadel.enginekt.transform.result.json.JsonNode
 import graphql.nadel.enginekt.transform.result.json.JsonNodeExtractor
-import graphql.nadel.enginekt.util.AnyList
-import graphql.nadel.enginekt.util.AnyMap
-import graphql.nadel.enginekt.util.asJsonMap
 import graphql.nadel.enginekt.util.emptyOrSingle
 import graphql.nadel.enginekt.util.flatten
-import graphql.nadel.enginekt.util.mapFrom
+import graphql.nadel.enginekt.util.makeNormalizedInputValue
+import graphql.nadel.enginekt.util.toMapStrictly
+import graphql.nadel.enginekt.util.javaValueToAstValue
 import graphql.normalized.ExecutableNormalizedField
 import graphql.normalized.NormalizedInputValue
-import graphql.schema.GraphQLArgument
-import graphql.schema.GraphQLTypeUtil
 
-internal typealias AnyAstValue = Value<*>
+internal class NadelHydrationInputBuilder private constructor(
+    private val instruction: NadelHydrationFieldInstruction,
+    private val aliasHelper: NadelAliasHelper,
+    private val fieldToHydrate: ExecutableNormalizedField,
+    private val parentNode: JsonNode,
+) {
+    companion object {
+        fun getInputValues(
+            instruction: NadelHydrationFieldInstruction,
+            aliasHelper: NadelAliasHelper,
+            fieldToHydrate: ExecutableNormalizedField,
+            parentNode: JsonNode,
+        ): List<Map<String, NormalizedInputValue>> {
+            return NadelHydrationInputBuilder(instruction, aliasHelper, fieldToHydrate, parentNode)
+                .build()
+        }
+    }
 
-internal object NadelHydrationInputBuilder {
-    fun getInputValues(
-        instruction: NadelHydrationFieldInstruction,
-        aliasHelper: NadelAliasHelper,
-        hydratedField: ExecutableNormalizedField,
-        parentNode: JsonNode,
-    ): List<Map<String, NormalizedInputValue>> {
-        val inputDefsForAllCalls = instruction.actorInputValueDefs.asSequence()
-            .let { defs ->
-                when (val hydrationStrategy = instruction.hydrationStrategy) {
-                    is NadelHydrationStrategy.OneToOne -> defs
-                    is NadelHydrationStrategy.ManyToOne -> defs.filter { def -> def != hydrationStrategy.inputDefToSplit }
-                }
-            }
+    private fun build(): List<Map<String, NormalizedInputValue>> {
```

Comment:
Ok I ended up refactoring this code while I was at it because I was never happy with it. I think I am now.

I suggest reading the class by itself instead of looking at the diff. It's almost completely changed.